doc/1.4/language.md

```diff
@@ -3396,114 +3396,79 @@ forms of `after` statements, the input parameters of the method are never
 required to be of serializable type, meaning pointers can be passed as arguments
 to the callback. But **beware**: pointers to stack-allocated data (pointers to
 or within `local` variables) must **never** be passed as arguments. The
-stack with which the `after` statement is executed is _not_ preserved, so any
+stack with which the `after` statement is executed is *not* preserved, so any
 pointers to stack-allocated data will point to invalid data by the time the
 callback is called. The DML compiler has some checks in place to warn about the
 most obvious cases where pointers to stack-allocated data are provided as
 arguments, but it is unable to detect all cases. It is ultimately the modeller's
 responsibility to ensure it doesn't happen.
 
 To detail a scenario exemplifying the kind of issues that immediate after may
-be leveraged to solve, consider the following device, representing an i2c master
-connected to a slave via `i2c_link`, and receiving responses from the slave via
-the `i2c_in` port. Its function is simple: once prompted, the device will start
-the slave, and then continually read packets from it until done, handling the
-asynchronous logic through an FSM.
-```
-connect i2c_link {
-    interface i2c_slave_v2;
-}
-
-port i2c_in {
-    implement i2c_master_v2 {
-        method acknowledge(i2c_ack_t ack_value) {
-            main_fsm.on_acknowledge();
-        }
-        method read_response(uint8 response) {
-            main_fsm.on_acknowledge(response);
-        }
+be leveraged to solve, consider the following device, which needs to communicate
+with a *manager* device and receive permission in order to perform a particular
+action. Its function is simple: once prompted, the device will raise a signal to
+the manager in order to request permission, and waits for it to respond with an
+acknowledgement. Once received, the device lowers the signal to the manager and
+performs the action it just received permission for.
+In order to implement the asynchronous logic needed for this, a simple FSM is
+used.
+```
+param STATE_IDLE = 0;
+param STATE_EXPECTING_ACK = 1;
+saved int curr_state = STATE_IDLE;
+
+connect manager {
+    interface signal;
+}
+
+implement signal {
+    method signal_raise() {
+        on_acknowledgement();
```

Comment:
consider grouping these two in a `port`. Implementing interfaces on top level is generally discouraged.

Reply:
Done.